src/lib/rnp.cpp

```diff
@@ -487,9 +492,16 @@ try {
         goto done;
     }
 
-    ob->key_provider = (pgp_key_provider_t){.callback = ffi_key_provider, .userdata = ob};
+    ob->key_provider =
+#ifndef _MSC_VER
+      (pgp_key_provider_t)
```

Comment:
Just curious - which error does this throw? Seems to be legit code.

Reply:
This is legit C code and MSVC compiles it only if the extension is `.c`.
So we either have to move it out to `.c ` files or ... like this

Reply:
Another way is to change it to the following:
```
ob->key_provider.callback = ffi_key_provider;
ob->key_provider.userdata = ob;
```
which would do the same and take less code lines/look cleaner.